File: src/leetcode/isAnagram.jl
"""
    isAnagram(s::String, t::String) -> Bool

Determines if two strings `s` and `t` are anagrams of each other.

# Arguments
- `s::String`: The first string.
- `t::String`: The second string.

# Returns
- `Bool`: `true` if `s` and `t` are anagrams, `false` otherwise.

# Example
```julia
isAnagram("listen", "silent") # returns true
isAnagram("hello", "world")   # returns false
```
"""

function isAnagram(s::String, t::String)::Bool
    s_dict = Dict{Char, Int}()
    t_dict = Dict{Char, Int}()

    for char in s
        s_dict[char] = get(s_dict, char, 0) + 1
    end

    for char in t
        t_dict[char] = get(t_dict, char, 0) + 1
    end

    return s_dict == t_dict
end

File: src/leetcode/is_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_dict = {}
        t_dict = {}
        for char in s:
            s_dict[char] = s_dict.get(char, 0) + 1
        for char in t:
            t_dict[char] = t_dict.get(char, 0) + 1
        return s_dict == t_dict


# Example usage
solution = Solution.isAnagram("anagram", "nagaram")
print(solution)  # Output: True
solution = Solution.isAnagram("rat", "car")
print(solution)  # Output: False
